src/components/Form/Form.tsx

```diff
@@ -116,11 +143,24 @@ const Form = <T extends FormDetails>({
               label={props.cancelLabel ?? "Cancel"}
             />
             <Submit
+              id="save"
               data-testid="submit-button"
               type="submit"
-              disabled={disabled}
-              label={props.submitLabel ?? "Submit"}
+              disabled={disabled || !isDirty}
+              label={props.submitLabel}
             />
+            {props.additionalButtons?.map((btn) => {
+              if (btn.label) {
+                return (
+                  <Submit
+                    id={btn.id}
+                    type={btn.type}
+                    disabled={disabled || !isDirty}
+                    label={btn.label}
+                  />
+                );
+              }
```

Comment:
1. why have an if block checking if label is present? the type definition says it's not nullable.
2. forgot to pass `key` ?

Reply:
I did removed it.
I think it would give an additional check